source/class.cpp

```diff
@@ -1091,6 +1093,10 @@ string ClassBinder::bind_nested_classes(Context &context)
 			b.bind(context);
 			c += b.code();  c += '\n';
 			prefix_code_ += b.prefix_code();
+			// make context aware of innerC
+			// (this registers innerC with context with C as a dependency)
+			BinderOP bop = std::shared_ptr<Binder>( new Dependency( innerC, C ) );
+			context.add( bop );
```

Comment:
This is problematic: right now Binder design is assume that there is a clear separation of parsing phase and generation phase. So allowing generation phase to add new Binders will break this. 

Let me think if there is a better way to handle this situation which preserve clear separation of parsing and generation phases

Reply:
Ok. That was not clear to me (since context does provide the `add` method.)